Mathlib/LinearAlgebra/BilinearMap.lean

```diff
@@ -119,6 +119,24 @@ def flip (f : M →ₛₗ[ρ₁₂] N →ₛₗ[σ₁₂] P) : N →ₛₗ[σ₁
     -- It looks like we now run out of assignable metavariables.
     (fun c n  m  => by simp only [map_smulₛₗ _, smul_apply])
 
+section lcomp
+
+variable (S N) [Module R N] [SMulCommClass R S N]
+
+/-- Composing a given linear map `M → N` with a linear map `N → P` as a linear map from
+`Nₗ →ₗ[R] Pₗ` to `M →ₗ[R] Pₗ`. -/
+def lcomp (f : M →ₗ[R] M₂) : (M₂ →ₗ[R] N) →ₗ[S] M →ₗ[R] N :=
+  flip <| LinearMap.comp (flip id) f
```

Comment:
Can you just upgrade `LinearMap.llcomp` to be heterobasic instead?

Reply:
Can you write the statement that you have in mind? Do you want three or four different rings (or ring homs for semilinear maps)? If you can make the statement compile I can probably figure out what SMulCommClasses are needed to make it true.

Reply:
I don't know how general you want but at least this subsumes both `LinearMap.llcomp` and your `lcomp` (modulo maybe the `CommSemiring` but should be fixable).
```
import Mathlib

def LinearMap.llcomp2 (R S : Type*) [CommSemiring R] [Semiring S] (M : Type*) (Nₗ : Type*) (Pₗ : Type*)
    [AddCommMonoid M] [AddCommMonoid Nₗ] [AddCommMonoid Pₗ] [Module R M] [Module R Nₗ] [Module R Pₗ] [Module S Pₗ]
    [SMulCommClass R S Pₗ] :
    (Nₗ →ₗ[R] Pₗ) →ₗ[S] (M →ₗ[R] Nₗ) →ₗ[R] M →ₗ[R] Pₗ where
  toFun := LinearMap.llcomp R M Nₗ Pₗ
  map_add' _ _ := rfl
  map_smul' _ _ := rfl
```

Reply:
I do want to apply it to a noncommutative (semisimple) R. Your version is a generalization of the bilinear map `llcomp`, and I don't think it should replace the linear map `lcomp` (and it can't unless you move `M →ₗ[R] Nₗ` to the front using some `LinearMap.flip`). If you have further suggestions I think they'd better come in the form of a commit.

Reply:
These are removed from this PR since they have been added by #26458.